### DISCUSSION TITLE
[Android] User session between the app and a service

### DISCUSSION THREAD

**Discussion by xieciun** [supabase-community/supabase-kt#714] [Q&A] [answered]
Hi,
I am building an app with the NotificationListener Service. I can see that after the user logs in (on the application level),  auth.currentSessionOrNull() returns the proper values when started from the app, but returns Null when started from the service. How can I make sure that the service can access the session? Should I pass the Refresh Token and use auth.refreshSession(refreshToken) or is there any configuration that allows me to use the same session across my app and the service?

**Comment by jan-tennert** [comment]
By default, the session gets loaded in when the app starts (and automatic session refresh gets started) and gets cleared when the app switches to the background. This is done to prevent expired sessions being still loaded in when coming back from background. You could disable this behavior via `AuthConfig#enableLifecycleCallbacks`, but then you will have to do that yourself. See the [Android implementation](https://github.com/supabase-community/supabase-kt/blob/master/GoTrue%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fgithub%2Fjan%2Fsupabase%2Fgotrue%2FsetupPlatform.kt#L29) for details.

**Reply by jan-tennert** [reply, marked as answer]
Maybe just using `Auth#loadFromStorage()` could also work

**Reply by xieciun** [reply]
Hi, yes it worked, when the app was active, but failed when the app was in the background. calling leadFromStorage solved it. Thnk you :)

**Comment by xieciun** [comment]
If anyone else has a similar issue: The problem was not in the refreshing session mechanism, but in the fact that the NotificationService can be run in a separate process than the main application. This means that load from storage will not access stored session information as SharedPrefferences are specific per process. To solve this, the Supabase logic need to be moved to WorkManager (called from the service). Work Manager is started within the Application process so it can access the session information.
